### PR TITLE
Disable h/w exceptions for coreclr (non-DAC) builds.

### PR DESCRIPTION
**Pull request by mikem8361**
Fixes the recursive GC crash.

Also fixed sos "clrstack -i". The sos data target needed to implement ICorDebugDataTarget4::VirtualUnwind too.

### CODE REVIEW

**Comment by mikem8361** [comment]
@janvorli could you review?

File: src/pal/inc/pal.h [hunk 1]
```diff
@@ -6870,6 +6870,7 @@ class NativeExceptionHolderFactory
     };                                                                          \
     try                                                                         \
     {                                                                           \
+        HardwareExceptionHolder                                                 \
```
Comment (janvorli) [inline src/pal/inc/pal.h]: I can see that you have removed the CatchHardwareExceptionHolder base class from the NativeExceptionHolderBase, but you have added it to the macros at all places where we have the native holder. So doesn't the change result the same behavior as before?

Comment (mikem8361) [inline src/pal/inc/pal.h]: No, “HardwareExceptionHolder” is a macro that is only active if FEATURE_ENABLE_HARDWARE_EXCEPTIONS is defined and it is defined only for DAC builds.

#ifdef FEATURE_ENABLE_HARDWARE_EXCEPTIONS
#define HardwareExceptionHolder CatchHardwareExceptionHolder __catchHardwareException;
#else
#define HardwareExceptionHolder
#endif // FEATURE_ENABLE_HARDWARE_EXCEPTIONS

Comment (janvorli) [inline src/pal/inc/pal.h]: Ah, I am sorry, I've missed that. Thanks for the explanation.

**Comment by janvorli** [comment]
LGTM

**Comment by mikem8361** [comment]
@dotnet-bot test OSX

**Comment by mikem8361** [comment]
@dotnet-bot test OSX x64 Debug Build

**Comment by mikem8361** [comment]
@dotnet-bot test OSX x64 Release Build

**Comment by mikem8361** [comment]
@dotnet-bot test OSX x64 Release Build and Test

**Pull request merged**